crates/pathfinder/src/state/block_hash.rs

```diff
@@ -0,0 +1,310 @@
+use anyhow::{Context, Error, Result};
+use bitvec::prelude::BitView;
+use stark_hash::{stark_hash, StarkHash};
+
+use crate::core::{SequencerAddress, StarknetBlockHash};
+use crate::sequencer::reply::{
+    transaction::{Event, Receipt, Transaction},
+    Block,
+};
+use crate::state::merkle_tree::MerkleTree;
+
+pub fn compute_block_hash(block: &Block) -> Result<StarknetBlockHash> {
```

Comment:
Could add some docs with pointers to what resources to read through when debugging a block hash verification failure, maybe to the python stuff as well.